Sources/Commands/Backend.swift

```diff
@@ -0,0 +1,160 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2018 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import TSCBasic
+import PackageModel
+import PackageGraph
+import Build
+import XCBuildSupport
+
+/// A protocol that represents a backend used by SwiftPM for all build, run and test operations. This allows factoring
+/// out the implementations details between SwiftPM's own backend and xcbuild on macOS.
+public protocol Backend {
```

Comment:
What about calling this protocol BuildSystemBackend or just BuildSystem?

Reply:
Done in next commit.